app/scripts/modules/core/src/application/nav/NavRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { UIRouterContextComponent } from '@uirouter/react-hybrid';
+import { UISref } from '@uirouter/react';
+
+import { NavCategory } from './NavCategory';
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../../application';
+
+export interface INavRouteProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavRoute = ({ app, category, isActive }: INavRouteProps) => (
+  <UIRouterContextComponent key={category.key}>
+    <UISref className={isActive ? 'active' : ''} to={category.sref}>
+      <a>
+        <NavCategory app={app} category={category} isActive={isActive} />
+      </a>
+    </UISref>
+  </UIRouterContextComponent>
+);
```

Comment:
ICYMI, there are now uirouter/react hooks that are (IMHO) slightly less awkward than `UISref`:
```suggestion
export const NavRoute = ({ app, category, isActive }: INavRouteProps) => {
  const sref = useSrefActive(category.sref, null, 'active');
  return (
    <UIRouterContextComponent key={category.key}>
      <a {...sref}>
        <NavCategory app={app} category={category} isActive={isActive} />
      </a>
    </UIRouterContextComponent>
  );
};
```

Reply:
Sweet, I had no idea. I will use this and remove the `UIRouterContextComponent`